scp/cc/cpio/client_providers/private_key_fetcher_provider/src/azure/azure_private_key_fetcher_provider_utils.cc

```diff
@@ -29,16 +29,326 @@ using google::scp::core::Uri;
 
 namespace google::scp::cpio::client_providers {
 
+bool AzurePrivateKeyFetchingClientUtils::isPrivate(EVP_PKEY* pkey) {
+  // Determine if the key is private or public
+  int key_type = EVP_PKEY_type(pkey->type);
+  bool is_private = false;
+  if (key_type == EVP_PKEY_RSA) {
+    const RSA* rsa_key = EVP_PKEY_get1_RSA(pkey);
+    is_private =
+        (rsa_key->e != NULL && rsa_key->n != NULL && rsa_key->d != NULL);
+  }
+  return is_private;
+}
+
 void AzurePrivateKeyFetchingClientUtils::CreateHttpRequest(
     const PrivateKeyFetchingRequest& request, HttpRequest& http_request) {
   const auto& base_uri =
       request.key_vending_endpoint->private_key_vending_service_endpoint;
   http_request.method = HttpMethod::POST;
 
   http_request.path = std::make_shared<Uri>(base_uri);
+
+  // Generate attestation report
   const auto report =
       hasSnp() ? fetchSnpAttestation() : fetchFakeSnpAttestation();
   CHECK(report.has_value()) << "Failed to get attestation report";
-  http_request.body = core::BytesBuffer(nlohmann::json(report.value()).dump());
+
+  nlohmann::json json_obj;
+  json_obj[kAttestation] = report.value();
+
+  http_request.body = core::BytesBuffer(json_obj.dump());
+}
+
+/**
+ * @brief Generate a new wrapping key
+ */
+std::pair<std::shared_ptr<EvpPkeyWrapper>, std::shared_ptr<EvpPkeyWrapper>>
+AzurePrivateKeyFetchingClientUtils::GenerateWrappingKey() {
```

Comment:
We need tests for these function

Reply:
test are implemented
https://github.com/KenGordon/data-plane-shared-libraries/blob/beejones/key-wrapping/scp/cc/cpio/client_providers/private_key_fetcher_provider/test/azure/azure_private_key_fetcher_provider_utils_test.cc